.github/workflows/build.yml

```diff
@@ -0,0 +1,68 @@
+name: Build and Test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        swift: ["5.2.1-bionic", "5.1.5-bionic", "4.2.4"]
+        # master as of 4/4/2020: cf601047ebf87cf7f443753ded41132eb689cb10
+        protobuf_gitref: ["v3.11.4", "cf601047ebf87cf7f443753ded41132eb689cb10"]
```

Comment:
> So we just update that one hash when we want to test a newer version of master? Works for me.

Yes, though this was a temporary workaround until I got an answer here: https://github.com/actions/checkout/issues/209
With that I will be able to put `"master"` in this list and make the caching safe.

Reply:
Is it worth landing that fix now while @tbkka works on getting the hook set up?

Reply:
Yup, I'll work on it.

Reply:
Issue 993 is a newer failure on updated conformance tests, so it would be nice to get our master validation update to float with those commits.

Reply:
@toffaletti how do we check if the caching of things is working?  Looking at https://github.com/apple/swift-protobuf/pull/994/checks it seems like it is building protobuf, I would have thought it would be cached from when @tbkka set things up and did a test PR to ensure they built?

Reply:
I’ll take a look.

Reply:
I agree it seems like the cache should have had a cache hit, in 
https://github.com/apple/swift-protobuf/runs/664761605?check_suite_focus=true I see
> Cache not found for input keys: Linux-ubuntu-18.04-protobuf-cf601047ebf87cf7f443753ded41132eb689cb10.

then later in Post Cache:

> Cache saved successfully

And in the other parallel builds:

> Running JavaScript Action with default external tool: node12
> [warning]Cache already exists. Scope: refs/pull/991/merge, Key: Linux-ubuntu-18.04-protobuf-cf601047ebf87cf7f443753ded41132eb689cb10, Version: (null)


But then 3 days later in your PR a cache miss for the same key:

https://github.com/apple/swift-protobuf/runs/674773193?check_suite_focus=true

> Cache not found for input keys: Linux-ubuntu-18.04-protobuf-cf601047ebf87cf7f443753ded41132eb689cb10.


However, I do see at least one cache hit:

https://github.com/apple/swift-protobuf/runs/675011233?check_suite_focus=true

> Cache Size: ~182 MB (190677067 B)
> /bin/tar -xz -f /__w/_temp/90437206-2a9e-46a5-9320-7515799d57c5/cache.tgz -C /__w/swift-protobuf/swift-protobuf/protobuf
> Cache restored from key: Linux-ubuntu-18.04-protobuf-6935eae45c99926a000ecbef0be20dfd3d159e71

It seems like the cache isn't reliable. I will open an issue on https://github.com/actions/cache and update here.

Reply:
https://github.com/actions/cache/issues/310 - will see if they can investigate this better than I'm able to. Meanwhile I'll work on the improved caching for floating on master.